src/handlers/publish.rs

```diff
@@ -179,36 +182,34 @@ pub async fn handle_publish(
         publish_index_file(package_entry).await?;
     }
 
```

Comment:
Should we move the publish to index file section into the db transaction scope, i.e couple lines below? If it is outside do we benefit from the transaction logic?

Reply:
I guess a cleaner rephrase of my question is: if publish index file fails do we rollback the db write if they stay like this?

Reply:
Good point; IMO it should be rolled-back, seems like the perfect use-case for this functionality.

Reply:
I agree, that was my original intent. The issue is that the diesel `transaction` can't accept an async function. I'm thinking of doing a deeper dive as a follow up.